task/show-sbom/0.1/show-sbom.yaml

```diff
@@ -10,35 +10,65 @@ metadata:
     tekton.dev/tags: "appstudio, hacbs"
 spec:
   description: >-
-    Shows the Software Bill of Materials (SBOM) generated for the built image in CyloneDX JSON format.
+    Shows the Software Bill of Materials (SBOM) generated for the built image in CyloneDX JSON format:
   params:
     - name: IMAGE_URL
       description: Fully qualified image name to show SBOM for.
       type: string
   steps:
   - name: show-sbom
-    image: quay.io/redhat-appstudio/cosign:v2.1.1@sha256:c883d6f8d39148f2cea71bff4622d196d89df3e510f36c140c097b932f0dd5d5
+    image: quay.io/redhat-appstudio/appstudio-utils:3e548a38b3ad183262a25bc2a4eb6b5367b83fb5
     # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
     # the cluster will set imagePullPolicy to IfNotPresent
     # also per direction from Ralph Bean, we want to use image digest based tags to use a cue to automation like dependabot or renovatebot to periodially submit pull requests that update the digest as new images are released.
     env:
     - name: IMAGE_URL
       value: $(params.IMAGE_URL)
     script: |
-      #!/busybox/sh
-      status=-1
-      max_try=5
-      wait_sec=2
-      for run in $(seq 1 $max_try); do
-        status=0
-        cosign download sbom $IMAGE_URL 2>>err
-        status=$?
-        if [ "$status" -eq 0 ]; then
-          break
-        fi
-        sleep $wait_sec
-      done
-      if [ "$status" -ne 0 ]; then
+      #!/bin/bash
+      IMAGE_URL=$1
+
+      download_sbom_with_retry() {
+        status=-1
+        max_try=5
+        wait_sec=2
+
+        PLATFORM_ARG="$1"
+        for run in $(seq 1 $max_try); do
+          status=0
+          cosign download sbom $PLATFORM_ARG $IMAGE_URL 2>>err
+          status=$?
+          if [ "$status" -eq 0 ]; then
+            break
+          fi
+          sleep $wait_sec
+        done
+        if [ "$status" -ne 0 ]; then
           echo "Failed to get SBOM after ${max_try} tries" >&2
           cat err >&2
+        fi
+      }
+      echo "Inspecting image ${IMAGE_URL}"
+
+      RAW_OUTPUT=$(skopeo inspect --no-tags --raw docker://${IMAGE_URL})
+      if [ $(jq -r '.mediaType' <<< $RAW_OUTPUT) == "application/vnd.oci.image.manifest.v1+json" ] ; then
+        ARCHES=""
+      else
+        # Multi arch
+        ARCHES=$(jq -r '.manifests[].platform.architecture' <<< $RAW_OUTPUT)
+      fi
+
+      if [ -z "${ARCHES}" ] ; then
+        # single arch image
+        echo ""
+        echo "Single arch image"
+        echo ""
+        download_sbom_with_retry ""
+      else
+        for arch in $ARCHES; do
+          echo ""
+          echo "Arch: $arch"
+          echo ""
+          download_sbom_with_retry " --platform=linux/$arch "
+        done
```

Comment:
Since this is just a task to show the summary in the console, I don't know if users would get any value iterating through all of the arches and showing _all_ SBOMs. I wouldn't expect this to be a primary means to see the information. Instead, this might decrease the utility as it just increases the size of the logs.

Reply:
See @ralphbean 's comment:

> https://github.com/redhat-appstudio/build-definitions/pull/871#issuecomment-1991902775

Reply:
Yes, the SBOM content is important, but I don't know if it is important in _this_ context. This Tekton task is just an "quick method" to show users what is in the builds. I posit that if users want to see detailed information then they will likely try to either access it locally where they can target specific architectures or they will upload the SBOMs to some tool like guac.sh.